girder_worker_utils/transforms/girder_io.py

```diff
@@ -37,36 +40,68 @@ def __init__(self, *args, **kwargs):
             self.gc = None
 
 
+@six.add_metaclass(abc.ABCMeta)
 class GirderClientResultTransform(ResultTransform, GirderClientTransform):
     pass
 
 
-class GirderFileId(GirderClientTransform):
+@six.add_metaclass(abc.ABCMeta)
+class _GirderResourceDownload(GirderClientTransform):
+    def __init__(self, _id, **kwargs):
+        super(_GirderResourceDownload, self).__init__(**kwargs)
+        self._id = _id
+        self._local_path = None
+
+    def _repr_model_(self):
+        return "{}('{}')".format(self.__class__.__name__, self._id)
+
+    @property
+    def local_path(self):
+        if self._local_path is None:
+            self._local_path = os.path.join(tempfile.mkdtemp(), str(self._id))
+        return self._local_path
+
+    def cleanup(self):
+        shutil.rmtree(os.path.dirname(self.local_path), ignore_errors=True)
```

Comment:
This is going to remove the downloaded folder or file,  but not the tempdir that was created.